### ISSUE TITLE
layout,syntax: Specify horizontal vs vertical arrangement

### ISSUE THREAD

**Issue by alixander** [terrastruct/d2#185]
These should be scoped to containers. E.g. something like

```d2
arrangement: horizontal

a: {
  arrangement: horizontal
  ...
}
b: {
  arrangement: vertical
}
```

so `a` and `b` would be horizontal, the contents of `a` would be horizontal, and the contents of `b` would be vertical. 

not sure `arrangement` is the best keyword here. also for the value, other tools use `LR` and `TD`, which... is perhaps more descriptive, but less obvious. tbd

**Comment by prowave** [comment]
What about defining the relationship type in addition to the container layout?

`peer` is horizontal
`hierarchical` is vertical and the default

```
a: Node 1
b: Node 2
c: Node 3
d: Application

a <-> b <-> c {
  type: peer
}

d -> a 
d -> b
d -> c
```
 
Or some variant of the above that fits within your standards. :)

**Comment by alixander** [comment]
the ability to specify it on a relationship level is possible in TALA, since it's our custom engine and we can do anything, but I don't think that granularity is supported in dagre/graphviz/elk. it's a good idea, but we'll probably do the container one first since it's portable across layout engines. ty @prowave

**Comment by guybarnhartmagen** [comment]
I agree - this is an important feature, especially when having extensive diagrams

**Comment by alixander** [comment]
This should actually come after glob support. Because we want it to apply recursively.

e.g.

```d2
a: {
  # All descendants + this container will have vertical
  **.orientation: vertical
}
```
